library/utils.py

```diff
@@ -0,0 +1,107 @@
+import csv
+import json
+import os
+import urllib2
+
+from django.core.files.base import ContentFile
+from django.utils.translation import ugettext as _
+
+OPENLIBRARY_API_URL = 'https://openlibrary.org/api/books?'
+
+
+def to_unicode(text):
+    """Do its best to return an unicode string."""
+    if isinstance(text, unicode):
+        return text
+    else:
+        try:
+            text = text.decode('utf-8')
+        except UnicodeDecodeError:
+            text = text.decode('latin-1')
+        return text
+
+
+def fetch_from_openlibrary(isbn):
+    """Fetch notice from Open Library and return a ready to use dict.
+    Doc: https://openlibrary.org/dev/docs/api/books."""
+    args = {
+        'jscmd': 'data',
+        'format': 'json',
+    }
+    key = 'ISBN:{isbn}'.format(isbn=isbn)
+    args['bibkeys'] = key
+    args = '&'.join('{0}={1}'.format(k, v) for k, v in args.iteritems())
+    url = '{base}{args}'.format(base=OPENLIBRARY_API_URL, args=args)
+    content = read_url(url)
+    try:
+        data = json.loads(content).get(key)
+    except ValueError:
+        return
+    if not data:
+        return
+    url = data.get('cover', {}).get('medium')
+    if url:
+        cover = load_cover_from_url(url)
+    else:
+        cover = None
+    publishers = data.get('publishers', [])
+    if publishers:
+        publisher = publishers[0]['name']
+    notice = {
+        'isbn': isbn,
+        'title': data.get('title'),
+        'authors': ', '.join([a['name'] for a in data.get('authors', [])]),
+        'cover': cover,
+        'publisher': publisher
```

Comment:
Careful here, if there's no `publishers` line 48, then `publisher` won't be initialized.